pkg/scheduler/scheduler.go

```diff
@@ -745,14 +745,14 @@ func (s *Scheduler) establishEstimatorConnections() {
 }
 
 // patchBindingStatusCondition patches schedule status condition of ResourceBinding when necessary.
-func patchBindingStatusCondition(karmadaClient karmadaclientset.Interface, rb *workv1alpha2.ResourceBinding, newScheduledCondition metav1.Condition) error {
+func patchBindingStatusCondition(karmadaClient karmadaclientset.Interface, rb *workv1alpha2.ResourceBinding, newScheduledCondition metav1.Condition, ignoreErr bool) error {
 	klog.V(4).Infof("Begin to patch status condition to ResourceBinding(%s/%s)", rb.Namespace, rb.Name)
 
 	updateRB := rb.DeepCopy()
 	meta.SetStatusCondition(&updateRB.Status.Conditions, newScheduledCondition)
-	// Postpone setting observed generation until schedule succeed, assume scheduler will retry and
-	// will succeed eventually.
-	if newScheduledCondition.Status == metav1.ConditionTrue {
+	// If an error during scheduling could be ignored, the scheduling result would have been patched successfully.
```

Comment:
New comments can be added under the existing comments.

Reply:
Done.